rtsf-at-checkout-device-scale/driver/scaleDriver_test.go

```diff
@@ -0,0 +1,279 @@
+			if tt.wantErr {
+				require.Error(t, err)
+				return
+			}
+			require.NoError(t, err)
+			require.NotEmpty(t, got)
+
+		})
+	}
+}
+
+func TestScaleDriver_HandleReadCommands(t *testing.T) {
+
+	config := scale.Config{
+		PortName:        "/dev/tty.usbserial-test",
+		BaudRate:        9600,
+		DataBits:        7,
+		StopBits:        1,
+		MinimumReadSize: 1,
+		ParityMode:      2,
+		TimeOutMilli:    500,
+	}
+
+	testDevice := scale.InitializeMockDevice(&config)
+
+	tests := []struct {
+		name           string
+		scaleConnected bool
+		wantErr        bool
+	}{
+		{
+			name:           "valid case",
+			scaleConnected: true,
+			wantErr:        false,
+		},
+		{
+			name:           "scale not connected",
+			scaleConnected: false,
+			wantErr:        true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			drv := getDefaultScaleDriver()
+			drv.scaleConnected = tt.scaleConnected
+			drv.scaleDevice = &scaleDevice{
+				serialDevice: testDevice,
+			}
+
+			gotRes, err := drv.HandleReadCommands("testDeviceName",
+				map[string]models.ProtocolProperties{},
+				[]dsModels.CommandRequest{
+					{
+						DeviceResourceName: "testDeviceResourceName",
+					},
+				},
+			)
+			if tt.wantErr {
+				require.Error(t, err)
+				return
+			}
+			require.NoError(t, err)
+			require.NotNil(t, gotRes)
+		})
+	}
+}
+
+func Test_findSerialPort(t *testing.T) {
+
+	tests := []struct {
+		name     string
+		portInfo enumerator.PortDetails
+		want     string
+		wantErr  bool
+	}{
+		{
+			name: "valid case",
+			portInfo: enumerator.PortDetails{
+				Name:         "testDevice",
+				IsUSB:        true,
+				PID:          "6001",
+				VID:          "0403",
+				SerialNumber: "0123456",
+			},
+			want:    "testDevice",
+			wantErr: false,
+		},
+		{
+			name: "vid pid not found",
+			portInfo: enumerator.PortDetails{
+				Name:         "testDevice",
+				IsUSB:        true,
+				PID:          "0000",
+				VID:          "0000",
+				SerialNumber: "0123456",
+			},
+			want:    "",
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			var ports []*enumerator.PortDetails
+			ports = append(ports, &tt.portInfo)
+			got, err := findSerialPort(ports, "6001", "0403")
+			if tt.wantErr {
+				require.Error(t, err)
+				return
+			}
+			require.NoError(t, err)
+			assert.Equal(t, tt.want, got)
+		})
+	}
+}
+
+func TestScaleDriver_AddDevice(t *testing.T) {
+	config := scale.Config{
+		PortName:        "/dev/tty.usbserial-test",
+		BaudRate:        9600,
+		DataBits:        7,
+		StopBits:        1,
+		MinimumReadSize: 1,
+		ParityMode:      2,
+		TimeOutMilli:    500,
+	}
+
+	testDevice := scale.InitializeMockDevice(&config)
+
+	type args struct {
+		deviceName string
+		protocols  map[string]models.ProtocolProperties
+		adminState models.AdminState
+	}
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
```

Comment:
Didn't catch this last time, but wantErr is always true??? Should have valid case?